### DISCUSSION TITLE
[RFC] Reusable cssinjs content from SSR & css file

### DISCUSSION THREAD

**Discussion by zombieJ** [ant-design/ant-design#43674] [RFCs - archive]
# 📋 RFC

## Summary

When inline HTML & css file contains the style UI needed. Reuse it without create cssinjs style.

## Basic example

```tsx
<StyleProvider recycling>
  <App />
</StyleProvider>
```

## Motivation

In modern web app, SSR is common usage which means style is already generated on the server side. So the first mount of component no need to do the cssinjs generate process. Just reuse it.

## Detailed

1. SSR provide the mark of `<Style />` tag or style selector mark to tell what have already prepared.
2. For the client side, if mark as `recycling` check the style selector to get full list by token level.
3. Revert insert into cache instead of generate it.

## Drawbacks

Need StyleProvider config open to avoid SSR not fully provide style.

## Adoption strategy

Provided in `@ant-design/cssinjs`.

**Comment by MadCcc** [comment]
Is cache still inserted？

**Reply by MadCcc** [reply]
I realized that the style check could insert exsited style into cache before handling client style.

**Reply by zombieJ** [reply]
When exist inline style, cache will be keep empty but mark as existed.
